packages/react-server/src/ReactFizzServer.js

```diff
@@ -162,58 +162,32 @@ export type KeyNode = [
   string | number /* key */,
 ];
 
-const REPLAY_NODE = 0;
-const REPLAY_SUSPENSE_BOUNDARY = 1;
-const RESUME_ELEMENT = 2;
-const RESUME_SUSPENSE_BOUNDARY = 3;
-const RESUME_SLOT = 4;
+type ResumeSlots =
+  | null // nothing to resume
+  | number // resume with segment ID at the root position
+  | {[index: number]: number}; // resume with segmentID at the index
 
 type ReplaySuspenseBoundary = [
-  1, // REPLAY_SUSPENSE_BOUNDARY
   string | null /* name */,
   string | number /* key */,
-  Array<ResumableNode> /* children */,
+  Array<ReplayNode> /* keyed children */,
```

Comment:
Instead of storing the key on the node we could also turn this into two keyed objects. One of number keys and one for string keys.